Packs/FeedThreatFox/Integrations/FeedThreatFox/FeedThreatFox.py

```diff
@@ -0,0 +1,405 @@
+        return [EntityRelationship(entity_a=value, entity_a_type=demisto_ioc_type,
+                                   name=name,
+                                   entity_b=related_malware, entity_b_type=FeedIndicatorType.Malware,
+                                   brand='ThreatFox Feed', reverse_name=reverse_name).to_indicator()]
+    return []
+
+
+def validate_interval(interval: int) -> int:
+    """Validates that the given interval is in days between 1 to 7.
+
+    Raises:
+        DemistoException: If the interval is invalid.
+
+    Returns:
+        int: The interval, if it is valid.
+    """
+    if interval % 1440 != 0:  # 1440 is the number of minutes in a day
+        raise DemistoException("The fetch interval must be in whole days, between 1-7.")
+    elif interval > 10080:
+        raise DemistoException("The fetch interval must not be more than 7 days.")
+    return interval
+
+
+def threatfox_get_indicators_command(client: Client, args: dict[str, Any]) -> CommandResults:
+
+    search_term = args.get('search_term')
+    id = args.get('id')
+    hash = args.get('hash')
+    tag = args.get('tag')
+    malware = args.get('malware')
+    limit = args.get('limit')
+
+    query_type = check_args_for_query(args)
+
+    query = create_query(query_type, id, search_term, hash, tag, malware, limit=limit)
+
+    demisto.debug(f'{LOG_LINE} calling api with {query=}')
+    try:
+        result = client.get_indicators_request(query)
+    except Exception:
+        if 'malware' in query:  # if illegal malware is provided an 502 error response returns
+            demisto.error('Make sure the malware you entered in valid')
+        raise
+
+    query_status = result.get('query_status')
+    query_data = result.get('data')
+
+    if query_status != 'ok' and query_status:
+        raise DemistoException(f'failed to run command, {query_status=}, {query_data=}')
+
+    demisto.debug(f'{LOG_LINE} got indicators')
+
+    parsed_indicators = parse_indicators_for_get_command(result.get('data') or result)
+    human_readable = tableToMarkdown(name='Indicators', t=parsed_indicators,
+                                     headers=['ID', 'Value', 'Description', 'MalwareFamilyTags',
+                                              'AliasesTags', 'FirstSeenBySource', 'LastSeenBySource', 'ReportedBy',
+                                              'Tags', 'Confidence', 'Publications'], removeNull=True, is_auto_json_transform=True)
+
+    return CommandResults(readable_output=human_readable)
+
+
+def fetch_indicators_command(client: Client, with_ports: bool, confidence_threshold: int,
+                             create_relationship: bool, interval: int, tlp_color: str, last_run: dict):
+
+    now = datetime.now(pytz.utc)
+    days_for_query = int(interval / 1440)  # The interval is validated already in the main
+
+    if last_run:
+        last_successful_run = datetime.strptime(last_run["last_successful_run"], '%Y-%m-%dT%H:%M:%SZ').replace(tzinfo=pytz.utc)
```

Comment:
Why not using dateparser?

Reply:
This ia the only way I maneged parsing the date in a way I would be able to do: time_delta = now - last_successful_run